src/validation/accommodationValidation.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable require-jsdoc */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-console */
+import Joi from '@hapi/joi';
+import Response from '../utils/response';
+
+/**
+ * @class accommodationValidator
+ */
+export default class accommodationValidator {
+  /**
+   * @param {object} req  request.
+   * @param {object} res  response.
+   * @param {function} next next.
+   * @returns {object}.
+   */
+  static async validateCreateAccommodation(req, res, next) {
+    const schema = Joi.object().keys({
+      name: Joi.string()
+        .min(2)
+        .trim()
+        .required()
+        .error(() => 'name field is required and must be a string'),
+      locationId: Joi.number()
```

Comment:
you can add .integer to exclude floating-point numbers but also the error message directs a user to enter a string check it out  
Also the key is locationId yet error message indicates location